Navigate to edit workflow after save

jyeshe opened the pull request:
## Notes for the reviewer

On refresh the new page is remounted (workflow is null) and the workflow is not on the query params.

The solution despite of that, was navigate to the edit mode after `save`, which makes sense in any case.

## Related issue

Fixes #1280 

## Review checklist

- [x] I have performed a **self-review** of my code
- [ ] I have verified that all appropriate **authorization policies** have been implemented and tested
- [ ] If needed, I have updated the **changelog**
- [ ] Product has **QA'd** this feature

jyeshe commented:
Missing to prevent navigation when already on edit.

taylordowns2000 commented:
@jyeshe , would you mind rebasing and marking this as not "draft" once it's ready for review?

elias-ba reviewed (changes requested):
Hey @jyeshe it feels like this is introducing a regression. When I open a workflow and I edit a job, I can no longer save it. Here is the error I am getting.
![image](https://github.com/OpenFn/Lightning/assets/86059666/b95ed034-5442-4acf-a7dd-b84760f646e0)

jyeshe commented:
> @jyeshe , would you mind rebasing and marking this as not "draft" once it's ready for review?

@taylordowns2000, @elias-ba 

I have rebased and there is still this problem of saving after refresh which I am working on. 
 
I was analyzing if it wouldn't be better fix it after #1244 however if we stay on the `:new` workflow mode there would be two things to fix:
1. After refresh the workflow is rendered as the initial one (only with 1 common job), as you've pointed out on the issue
2. Trying to save it gives an error of name uniqueness (the name is preserved but sets a new id).

I have switched back to `push_navigate` once with `push_patch` the panels stop working and I think switching to `:edit` mode is a coherent move. Working on the `Repo.update`'s changeset after refresh ...

taylordowns2000 reviewed (changes requested):
Getting some strange behavior here: https://www.loom.com/share/d5137003fb684d25bfe6197bc2d92f1c

Won't let me navigate to the new workflow

jyeshe commented:
> Getting some strange behavior here: https://www.loom.com/share/d5137003fb684d25bfe6197bc2d92f1c
> 
> Won't let me navigate to the new workflow

Maybe some cache @taylordowns2000 ? Could you try after a `mix do ecto.drop, ecto.setup` and clearing the cache. 
I can't reproduce it here and I have just rebased again the tests.
